### ISSUE TITLE
Query table contain LONG data type will failed with Lob pre-fetch option on

### ISSUE THREAD

**Issue by travelliu** [sijms/go-ora#331]
use 2.6.6 no data return

[go-ora-trace.log](https://github.com/sijms/go-ora/files/11059565/go-ora-trace.log)

use 2.5.16 normal

**Comment by travelliu** [comment]
![2023-03-24 at 16 41](https://user-images.githubusercontent.com/7792329/227468576-3a754515-99c2-411c-8c24-7cecf19990a8.png)

pre lob issue

![2023-03-24 at 16 42](https://user-images.githubusercontent.com/7792329/227468674-e1e7a790-397c-41f4-9aee-d65b474659e2.png)
Lob default=0

**Comment by sijms** [comment]
the problem is in  [command.go](https://github.com/sijms/go-ora/blob/cfc2b95292a67753631b5adfa5fa95da6c282854/v2/command.go#L1760)
because the sql query contain LONG datatype
```golang
// add stmt._hasLONG will fix the issue
if stmt._hasBLOB || stmt._hasLONG {
	if stmt.connection.connOption.Lob == 0 {
		stmt.define = true
		stmt.execute = false
		stmt.parse = false
		stmt.reSendParDef = false
		err = stmt.queryLobPrefetch(stmt.getExeOption(), dataSet)
		if err != nil {
			return nil, err
		}
	} else {
		err = stmt.readLobs(dataSet)
		if err != nil {
			return nil, err
		}
	}
}
```

**Comment by sijms** [comment]
fixed in v2.6.7

**Comment by travelliu** [comment]
2.6.7 long data errpr

```
SELECT PARTITION_NAME, HIGH_VALUE, TABLESPACE_NAME, COMPOSITE FROM DBA_TAB_PARTITIONS WHERE TABLE_OWNER = :1 AND TABLE_NAME = :2 ORDER BY PARTITION_POSITION
&sql.ColumnType{name:"PARTITION_NAME", hasNullable:true, hasLength:true, hasPrecisionScale:false, nullable:true, length:128, databaseType:"NCHAR", precision:0, scale:0, scanType:(*reflect.rtype)(0x1007e5780)}
&sql.ColumnType{name:"HIGH_VALUE", hasNullable:true, hasLength:false, hasPrecisionScale:false, nullable:true, length:0, databaseType:"LONG", precision:0, scale:0, scanType:(*reflect.rtype)(0x1007e5780)}
&sql.ColumnType{name:"TABLESPACE_NAME", hasNullable:true, hasLength:true, hasPrecisionScale:false, nullable:true, length:30, databaseType:"NCHAR", precision:0, scale:0, scanType:(*reflect.rtype)(0x1007e5780)}
&sql.ColumnType{name:"COMPOSITE", hasNullable:true, hasLength:true, hasPrecisionScale:false, nullable:true, length:3, databaseType:"NCHAR", precision:0, scale:0, scanType:(*reflect.rtype)(0x1007e5780)}
YR0 T USERS 5553455253
YR7 T USERS 5553455253
YR8 T USERS 5553455253
YR9 M USERS 5553455253
```

![2023-03-25 at 10 00](https://user-images.githubusercontent.com/7792329/227680175-d6bd421c-3a15-4165-be42-4189889761fc.png)


```
-- ORA_MTK.RANGE_PART definition

CREATE TABLE "ORA_MTK"."RANGE_PART" 
   (	"PROF_HISTORY_ID" NUMBER(10,0), 
	"PERSON_ID" NUMBER(10,0) NOT NULL ENABLE, 
	"ORGANIZATION_ID" NUMBER(10,0) NOT NULL ENABLE, 
	"RECORD_DATE" DATE NOT NULL ENABLE
   ) PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
  STORAGE(
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "USERS" 
  PARTITION BY RANGE ("RECORD_DATE") 
 (PARTITION "YR0"  VALUES LESS THAN (TO_DATE(' 2017-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS', 'NLS_CALENDAR=GREGORIAN')) SEGMENT CREATION DEFERRED 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING 
  STORAGE(
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "USERS" , 
 PARTITION "YR7"  VALUES LESS THAN (TO_DATE(' 2018-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS', 'NLS_CALENDAR=GREGORIAN')) SEGMENT CREATION DEFERRED 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING 
  STORAGE(
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "USERS" , 
 PARTITION "YR8"  VALUES LESS THAN (TO_DATE(' 2019-01-01 00:00:00', 'SYYYY-MM-DD HH24:MI:SS', 'NLS_CALENDAR=GREGORIAN')) SEGMENT CREATION DEFERRED 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING 
  STORAGE(
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "USERS" , 
 PARTITION "YR9"  VALUES LESS THAN (MAXVALUE) SEGMENT CREATION IMMEDIATE 
  PCTFREE 10 PCTUSED 40 INITRANS 1 MAXTRANS 255 
 NOCOMPRESS LOGGING 
  STORAGE(INITIAL 8388608 NEXT 1048576 MINEXTENTS 1 MAXEXTENTS 2147483645
  PCTINCREASE 0 FREELISTS 1 FREELIST GROUPS 1
  BUFFER_POOL DEFAULT FLASH_CACHE DEFAULT CELL_FLASH_CACHE DEFAULT)
  TABLESPACE "USERS" ) ;
```

**Comment by sijms** [comment]
yes there is a problem getting LONG columns in lob prefetch mode

**Comment by sijms** [comment]
I test the query with c# driver and give same result
would you run this query on another golang driver and share the network starting from the query

**Comment by sijms** [comment]
actually C# driver building a query to read long data type
```c#
try
{
    string empty = string.Empty;
    dtUnmarshaller = (DataUnmarshaller) null;
    string commandText;
    OracleDataReaderImpl fetchLongData = this.ExecuteQueryToFetchLongData(connection, currentRow, columnIndex, length, out commandText);
    TTCLongAccessor ttcLongAccessor = (TTCLongAccessor) null;
    if (fetchLongData != null && fetchLongData.m_accessors != null)
    {
        if (fetchLongData.m_accessors[0] is TTCLongAccessor)
        {
            ttcLongAccessor = fetchLongData.m_accessors[0] as TTCLongAccessor;
            currentRow = 0;
            columnIndex = 0;
            dtUnmarshaller = fetchLongData.m_dataUnmarshaller;
            fetchLongData.m_dataUnmarshaller = (DataUnmarshaller) null;
        }
        fetchLongData.ReleaseCursor(commandText);
    }
}
```
so I think there is a problem reading long data in lob pre-fetch mode

**Comment by sijms** [comment]
fixed in version 2.6.8
* know you can return up to 0x3FFFFFFF of data from long coumn type
* examples/long insert 0x3FFF bytes of data into long column and query it again
* for large data size better use `lob fetch=post`
the code is tested against oracle server versions: 12c and 19c
